### PR TITLE
Upgrade dependencies, including tslint. Also fix a ton of linting issues. 

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/compile/data/summary.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export namespace summary {
     /* dictionary mapping field name => dict set of aggregation functions */
     let meas: Dict<StringSet> = {};
 
-    model.forEach(function(fieldDef: FieldDef, channel: Channel) {
+    model.forEach(function(fieldDef: FieldDef, _: Channel) {
```
Comment (kanitw) [inline src/compile/data/summary.ts]: if last parameter isn't used, just remove it?

File: src/compile/data/timeunit.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ import {LayerModel} from '../layer';
 import {Model} from '../model';
 
 function parse(model: Model): Dict<VgFormulaTransform> {
-  return model.reduce(function(timeUnitComponent: Dict<VgFormulaTransform>, fieldDef: FieldDef, channel: Channel) {
+  return model.reduce(function(timeUnitComponent: Dict<VgFormulaTransform>, fieldDef: FieldDef, _: Channel) {
```
Comment (kanitw) [inline src/compile/data/timeunit.ts]: if last parameter isn't used, just remove it?

File: src/log.ts [hunk 3]
```diff
@@ -81,15 +81,15 @@ export function reset() {
   return current;
 }
 
-export function warn(...args: any[]) {
+export function warn(..._: any[]) {
```
Comment (kanitw) [inline src/log.ts]: This case the lint is absurd.  I strongly prefer ...args

File: tslint.json [hunk 4]
```diff
@@ -8,18 +8,17 @@
...
     "forin": true,
     "label-position": true,
-    "label-undefined": true,
     "no-arg": true,
     "no-any": false,
     "jsdoc-format": true,
+    "adjacent-overload-signatures": true,
     "member-access": true,
     "semicolon": [true, "always"],
-    "no-duplicate-key": true,
```
Comment (kanitw) [inline tslint.json]: is this one deprecated?

File: src/compile/data/summary.ts (hunk 1 above)
Comment (domoritz) [inline src/compile/data/summary.ts]: I left this particular one as it may be interesting but I'm okay with deleting.

File: src/log.ts (hunk 3 above)
Comment (domoritz) [inline src/log.ts]: It's actually not linting any more but enforced by the compiler. It's much better at picking up unused vars as you can see.

File: tslint.json (hunk 4 above)
Comment (domoritz) [inline tslint.json]: TS 2.0 checks this already.

File: src/compile/data/timeunit.ts (hunk 2 above)
Comment (domoritz) [inline src/compile/data/timeunit.ts]: Done

**Comment by domoritz** [comment]
Merge if you like it. Good night!

**Comment by kanitw** [comment]
Feel free to merge if Travis finishes.

**Pull request merged**